nomenclature/__init__.py

```diff
@@ -63,24 +64,55 @@ def parse_model_registration(
     """
     if not isinstance(output_directory, Path):
         output_directory = Path(output_directory)
-
-    region_aggregregation_mapping = RegionAggregationMapping.from_file(
+    # parse Common-Region-Mapping
+    region_aggregation_mapping = RegionAggregationMapping.from_file(
         model_registration_file
     )
     file_model_name = "".join(
         x if (x.isalnum() or x in "._- ") else "_"
-        for x in region_aggregregation_mapping.model[0]
+        for x in region_aggregation_mapping.model[0]
     )
-    region_aggregregation_mapping.to_yaml(
+    region_aggregation_mapping.to_yaml(
         output_directory / f"{file_model_name}_mapping.yaml"
     )
+    # parse Region-Country-Mapping
+    native = "Native region (as reported by the model)"
+    constituents = "Country name"
+    try:
+        region_country_mapping = pd.read_excel(
+            model_registration_file,
+            sheet_name="Region-Country-Mapping",
+            header=2,
+            usecols=[native, constituents],
+        )
+        region_country_mapping = region_country_mapping.where(
```

Comment:
I'm pretty sure that these three lines can be removed. In line 92, you drop all the na values anyway so the replacement with `None` does not have any effect. I quickly tested it locally and without those lines, the test still passes.
```suggestion
```